drivers/spi/spi_ll_stm32.c

```diff
@@ -971,17 +988,20 @@ static void spi_stm32_irq_config_func_##id(const struct device *dev)		\
 #define STM32_SPI_USE_SUBGHZSPI_NSS_CONFIG(id)
 #endif
 
+
+
 #define STM32_SPI_INIT(id)						\
 STM32_SPI_IRQ_HANDLER_DECL(id);						\
 									\
 PINCTRL_DT_INST_DEFINE(id);						\
 									\
+static const struct stm32_pclken pclken_##id[] =			\
+					       STM32_DT_INST_CLOCKS(id);\
+									\
 static const struct spi_stm32_config spi_stm32_cfg_##id = {		\
 	.spi = (SPI_TypeDef *) DT_INST_REG_ADDR(id),			\
-	.pclken = {							\
-		.enr = DT_INST_CLOCKS_CELL(id, bits),			\
-		.bus = DT_INST_CLOCKS_CELL(id, bus)			\
-	},								\
+	.pclken = pclken_##id,						\
+	.pclk_len = DT_INST_NUM_CLOCKS(id),				\
```

Comment:
one suggestion here:

similar to what we do in pinctrl, I'd create another struct that contains the array and size, so that drivers can simply do:

```
STM32_DT_INST_CLOCKS_DEFINE(id);
```

Then device stores a reference to the defined structure.

Reply:
Indeed. I though about it but wanted to see how the whole feature is used across drivers before that.
Would it be fine to deploy in a second step. I agree it makes the transition a little bit 'harder' tough.